analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CreatingHashAlgorithms.cs

```diff
@@ -1,77 +1,104 @@
-﻿namespace NS
-{
-    using System.Security.Cryptography;
+﻿using System.Security.Cryptography;
 
-    public class TestClass
+namespace Tests.Diagnostics
+{
+    public class InsecureHashAlgorithm
     {
-        // RSPEC 4790: https://jira.sonarsource.com/browse/RSPEC-4790
-        public void ComputeHash()
+        public void Hash(byte[] temp)
         {
-            // Review all instantiations of classes that inherit from HashAlgorithm, for example:
-            HashAlgorithm hashAlgo = HashAlgorithm.Create();
-//                                   ^^^^^^^^^^^^^^^^^^^^^^    {{Make sure this weak hash algorithm is not used in a sensitive context here.}}
-            HashAlgorithm hashAlgo2 = HashAlgorithm.Create("SHA1");
```

Comment:
With the new tracking we have, you can add one test to document variable tracking:
```
var name = "SHA512";
name = "SHA1";
HashAlgorithm.Create(name); // Noncompliant
name = "SHA512";
HashAlgorithm.Create(name); // Compliant

Reply:
We do have a similar case for MD5 and it's handled the same for all. I can change it a bit to check also a valid value but I would not add test cases for each algorithm since the code which is handling them is the same and they will grow a lot and will be hard to maintain.